crates/sui-rest-api/src/transactions/execution.rs

```diff
@@ -356,3 +352,135 @@ fn derive_balance_changes(
         })
         .collect()
 }
+
+pub struct SimulateTransaction;
+
+impl ApiEndpoint<RestService> for SimulateTransaction {
+    fn method(&self) -> axum::http::Method {
+        axum::http::Method::POST
+    }
+
+    fn path(&self) -> &'static str {
+        "/transactions/simulate"
+    }
+
+    fn operation(
+        &self,
+        generator: &mut schemars::gen::SchemaGenerator,
+    ) -> openapiv3::v3_1::Operation {
+        OperationBuilder::new()
+            .tag("Transactions")
+            .operation_id("SimulateTransaction")
+            .query_parameters::<SimulateTransactionQueryParameters>(generator)
+            .request_body(RequestBodyBuilder::new().bcs_content().build())
+            .response(
+                200,
+                ResponseBuilder::new()
+                    .json_content::<TransactionSimulationResponse>(generator)
```

Comment:
So this will create a json from the bcs? How do I interpret this?

Reply:
No this is how the OpenAPI spec is defined

Reply:
ah, got it. thanks!